department-of-veterans-affairs/abd-vro
# MCP-1989 V2 hypertension

## What was the problem?

Hypertension logic was deficient for determining the fast track eligibility for a claim. Not enough evidence was considered and the processor code needed more rules for eligibility. 

Associated tickets or Slack threads:

- [MCP-1989](https://amida.atlassian.net/browse/MCP-1989)
- https://github.com/department-of-veterans-affairs/abd-vro/issues/447
## How does this fix it?[^1]

Adds a queue for "7101v2" which responds with an object that includes a Boolean field for `sufficientForFastTracking` based on available data. The processor connected to the queue now considers "condition" evidence and has rules for determining eligibility for fast tracking. 

### notes: 
- changed `date_of_claim` to `dateOfClaim` because it should be returned to the VRO user as it is used in the decision process. The response body should be all camelCase. 
- Not many Ligththouse test veterans have recent BP data, it is difficult to find a positive case. 

## How to test this PR
- Step 1
- run `./gradlew build check docker`
- run `docker compose -f app/src/docker/docker-compose.yml -f app/src/docker/docker-compose.v2.yml up --build`
- Step 2
- go to swagger web page and run 
 `{
  "veteranIcn": "21000177",
  "diagnosticCode": "7101v2",
  "claimSubmissionId": "1234"
}`
[^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
